### ISSUE TITLE
Wrong friction?

### ISSUE THREAD

**Issue by kortenkamp** [CindyJS/CindyJS#530]
The attached example shows wrong behavior with a mass on a circle in physics simulation. It seems to be much too much friction in the system.

[Pendulum.zip](https://github.com/CindyJS/CindyJS/files/571395/Pendulum.zip)

**Comment by gagern** [comment]
This is not a problem with friction, since the file doesn't specify any friction at all, and the [default friction](https://github.com/CindyJS/CindyJS/blob/v0.8.2/src/js/liblab/LabObjects.js#L929) is zero, too. Instead this is a problem in our handling of semi-free objects. We'd need to project the actual force onto the tangential direction, which in this case would lead to a horizontal acceleration component. But we don't do that yet, so all the impulse is vertical, and the movement stops immediately once the mass has reached the lowest point of the circle. This is a known problem, which we [discussed in at least one meeting](https://github.com/CindyJS/CindyJS/wiki/Meeting-2016-09-19#physik-allgemein). Good to have a ticket for it, though.

**Comment by gagern** [comment]
I just wrote some code to this effect in [my tangentForce branch](https://github.com/gagern/CindyJS/compare/tangentForce). Although it's definitely a step in the right direction, there appears to be something missing since the mass will still come to rest pretty soon despite there being no friction. At the moment I'm not sure where to look.

Perhaps @richter-gebert has an idea of how the physics simulation in CindyJS differs from that in Cinderella to account for this difference? In particular, can I take the current position of the geometric element when computing forces, or do I have to use a different set of coordinates? I tried using `[beh.x, beh.y]` indtead, but that changed nothing.

**Comment by kortenkamp** [comment]
> Am 05.11.2016 um 13:50 schrieb Martin von Gagern <notifications@github.com>:
> 
> I just wrote some code to this effect in my tangentForce branch <https://github.com/gagern/CindyJS/compare/tangentForce>. Although it's definitely a step in the right direction, there appears to be something missing since the mass will still come to rest pretty soon despite there being no friction. At the moment I'm not sure where to look.
> 
> 

And this is not due to gravity?

Ulli

**Comment by gagern** [comment]
No, this is not due to gravity. In the absence of friction, gravity causes a continual conversion between potential and kinetic energy, but no dissipation of energy. Cinderella does reproduce this fairly well, CindyJS with my adjustment does not, and I don't know where the relevant difference lies.

**Comment by gagern** [comment]
I think I found the missing part: [this commented-out Java code here](https://github.com/CindyJS/CindyJS/blob/676a2e7e8f9eb26db324662686b89693b4f806cd/src/js/liblab/LabObjects.js#L186-L202) changes the direction of the current velocity while preserving its magnitude. I need to reproduce this in CindyJS, preferably in a way that works with the generic tangent direction concept.

**Comment by gagern** [comment]
I've added code to preserve the magnitude of the current velocity. Now I'm observing an increase of total energy in the example configuration (now included in one commit as well), and again I don't know where the relevant difference might be.
